torch/jit/annotations.py

```diff
@@ -288,10 +288,23 @@ def try_ann_to_type(ann, loc):
         elem_type = try_ann_to_type(ann.__args__[0], loc)
         if elem_type:
             return ListType(elem_type)
+        else:
+            warnings.warn("List must have a contained type")
     if is_dict(ann):
-        key = try_ann_to_type(ann.__args__[0], loc)
-        value = try_ann_to_type(ann.__args__[1], loc)
-        return DictType(key, value)
+        if ann.__args__:
+            key = try_ann_to_type(ann.__args__[0], loc)
+            value = try_ann_to_type(ann.__args__[1], loc)
+            if key and value:
+                return DictType(key, value)
+            else:
+                warnings.warn("Dict must have a key and value type")
+        else:
+            # If we don't return early here, the
+            # inspect.isclass(ann) branch below will execute
+            # in python3.6 and below, which will give the caller
```

Comment:
Nice find.

OOC, what changed at 3.6 to make this not happen any more?

Reply:
A `Dict` annotation in python3.6 without any contained types has an `__args__` attr but it is `None`. In 3.8, that attribute is a tuple of `TypeVars`.

Reply:
Nvm, found a simpler solution.